cpp-package/include/mxnet-cpp/ndarray.hpp

```diff
@@ -313,12 +313,22 @@ inline std::vector<mx_uint> NDArray::GetShape() const {
   return ret;
 }
 
+inline int NDArray::GetDType() const {
+  int ret;
+  MXNDArrayGetDType(blob_ptr_->handle, &ret);
+  return ret;
+}
+
 inline const mx_float *NDArray::GetData() const {
```

Comment:
@lx75249 let's change this to 
`
template<typename DType = mx_float>
inline const DType *NDArray::GetData() const {}